pfcpiface/telemetry.go

```diff
@@ -4,12 +4,24 @@
 package main
 
 import (
+	"log"
 	"net/http"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+var (
+	pfcpMessages = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "pfcp_messages_total",
+		Help: "Counter for incoming and outgoing PFCP messages",
+	}, []string{"message_type", "direction", "result"})
+	activeSessionCnt = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "active_current_sessions_total",
+		Help: "The total number of Users with active sessions in UPF",
+	})
+)
+
```

Comment:
Instead of being global variables convert into two types - 

connStats - to contain pfcpMessages and any future connection/message level metrics.
sessionStats - to contain activeSessionCnt and any future session manager level metrics

Add connStats member in PFCPConn
Add sessionStats member in PFCPSessionMgr

We should use Set method on activeSessionsCnt instead of Inc and Dec. It should be set with len(PFCPSessionMgr.sessions). This modification should entirely be contained within sessions.go

Reply:
this would limit the counter to particular connection and association. upf in the future might have multiple connections from different spgw/smf. so its better to keep it global right ?

Reply:
Then we need to include connection identifier into the flag.

Reply:
Also the updation of sessions need to happen from within sessions.go maybe we should add delete method there. And use set instead of Inc and Dec. What do you think?